src/Compiler/Service/FSharpCheckerResults.fs

```diff
@@ -259,9 +259,34 @@ type FSharpSymbolUse(denv: DisplayEnv, symbol: FSharpSymbol, inst: TyparInstanti
 
     member _.Range = range
 
+    member this.IsPrivateToFileAndSignatureFile =
+
+        let couldBeParameter, declarationLocation =
+            match this.Symbol with
+            | :? FSharpParameter as p -> true, Some p.DeclarationLocation
+            | :? FSharpMemberOrFunctionOrValue as m when not m.IsModuleValueOrMember -> true, Some m.DeclarationLocation
+            | _ -> false, None
+
+        let thisIsSignature = SourceFileImpl.IsSignatureFile this.Range.FileName
+
+        let signatureLocation =
+            match this.Symbol.Item with
+            | Item.Value v -> v.Deref.ArgReprInfoForDisplay
+            | _ -> None
+            |> Option.bind (fun a -> a.OtherRange)
```

Comment:
@0101 Would it be possible to extract some of this logic to `FSharpSymbol`?

Reply:
I think so. We can move `SignatureLocation` there. And possibly even the whole `IsPrivateToFileAndSignatureFile`.